From 42fda33f3b4b82cb1ecebcec731f52e9d79e26a7 Mon Sep 17 00:00:00 2001
From: "Apekshit(Appy) Sharma" <appy@cloudera.com>
Date: Wed, 22 Apr 2015 15:09:20 -0700
Subject: [PATCH] HBASE-13532 Log possible causes and fix in info message
 before UnknownScannerException is thrown in rpc call for scan. (Apekshit)

---
 .../java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java    | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java
index d09bd8d..1573e2f 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java
@@ -2173,7 +2173,11 @@ public class RSRpcServices implements HBaseRPCErrorHandler,
       if (request.hasScannerId()) {
         rsh = scanners.get(scannerName);
         if (rsh == null) {
-          LOG.info("Client tried to access missing scanner " + scannerName);
+          LOG.info("Client tried to access missing scanner " + scannerName
+            + ". This can happen if wrong scanner id is given or if scanner lease expired"
+            + " because of long wait between consecutive client checkins, or serverside"
+            + " closing down, or RS restart during upgrade. A possible fix is increasing"
+            + " the hbase.client.scanner.timeout.period configuration.");
           throw new UnknownScannerException(
             "Name: " + scannerName + ", already closed?");
         }
-- 
2.3.2 (Apple Git-55)

